internal/common/cloudproviderapi/client.go

```diff
@@ -193,6 +193,68 @@ func (c *Client) DeleteAWSCloudWatchScrapeJob(ctx context.Context, stackID strin
 	return nil
 }
 
+type AzureCredential struct {
```

Comment:
Should we model the `MetricCollectionConfiguration` at all or is it left off because it's part of the API response and terraform doesn't need to manage it/be aware of iit?